conseil-api/src/main/scala/tech/cryptonomic/conseil/api/routes/platform/data/ethereum/EthereumDataRoutesCreator.scala

```diff
@@ -0,0 +1,126 @@
+package tech.cryptonomic.conseil.api.routes.platform.data.ethereum
+
+import akka.http.scaladsl.server.Directives.concat
+import akka.http.scaladsl.server.Route
+import cats.instances.either._
+import cats.instances.future._
+import cats.syntax.bitraverse._
+import com.typesafe.scalalogging.LazyLogging
+import tech.cryptonomic.conseil.api.metadata.MetadataService
+import tech.cryptonomic.conseil.api.routes.platform.data.ApiDataRoutes
+import tech.cryptonomic.conseil.common.config.MetadataConfiguration
+import tech.cryptonomic.conseil.common.ethereum.EthereumTypes.EthereumBlockHash
+import tech.cryptonomic.conseil.common.generic.chain.DataTypes.QueryResponseWithOutput
+import tech.cryptonomic.conseil.common.metadata
+import tech.cryptonomic.conseil.common.metadata.{EntityPath, NetworkPath, PlatformPath}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/** Trait, which contains routes for Ethereum-related block-chains */
+trait EthereumDataRoutesCreator
+  extends EthereumDataHelpers
```

Comment:
I'm not sure: is this a trait extending a concrete class?

Reply:
I'm not sure what you are asking for? This trait is combined with Circe and Akka HTTP and indeed `EthereumDataHelpers` is a class. It is a class to cut off this long traits tail at some point but it could be a trait. I probably had different composition previously and did not revert it back.

Reply:
Sorry but I've never seen a concrete class being subclassed with a trait... I'm scared of possible consequences down the line 😅